NSS-Day-Cohort-68/rare-client-rare-team-4
# Adds page to view all posts

This pull request adds the functionality to view all posts in the database.

Testing info:
1. This test will require some test data. Run the following SQL lines to create the data in your database.
- Valid posts:
```sql
INSERT INTO Posts (user_id, category_id, title, publication_date, image_url, content, approved)
VALUES (1, 1, 'First Example Post', '2023-01-01 00:00:00', 'https://jooinn.com/images/beauty-of-nature-24.jpg', 'Lorem ipsum...', 1);

INSERT INTO Posts (user_id, category_id, title, publication_date, image_url, content, approved)
VALUES (2, 1, 'Second Example Post', '2023-01-02 00:00:00', null, 'Lorem ipsum...', 1);

INSERT INTO Posts (user_id, category_id, title, publication_date, image_url, content, approved)
VALUES (3, 1, 'Third Example Post', '2023-01-03 00:00:00', 'https://www.pixelstalk.net/wp-content/uploads/2016/07/Wallpapers-pexels-photo.jpg', 'Lorem ipsum...', 1);
```
<h6>NOTE: This test data involves having 3 users in your database. Errors may occur if you do not have at least 3 users, for this example.</h6>

- A post that has an _invalid publication date_:
```sql
INSERT INTO Posts (user_id, category_id, title, publication_date, image_url, content, approved)
VALUES (1, 1, 'Example Post from THE FUTURE', '2027-01-01 00:00:00', null, 'Lorem ipsum...', 1);
```
- A post that has _not been approved_:
```sql
INSERT INTO Posts (user_id, category_id, title, publication_date, image_url, content, approved)
VALUES (1, 1, 'Example UNAPPROVED post', '2023-01-01 00:00:00', null, 'Lorem ipsum...', 0);
```
2. Run the `server.py` file in the api directory to start the server.
3. Run `npm start` in the client directory to start the project.
4. You should see **All Posts** and **My Posts** navbar options.
5. Assuming you are logged in, clicking on **All Posts** should take you to a view in which you can see all the valid posts in the database.
6. These posts should include a `title`, `author`, and `category`.
7. If you used the test data provided above (and did not already have other posts in your database), then you should only see 3 posts.
    - The "Example Post from THE FUTURE" should _not_ be visible.
    - The "Example UNAPPROVED post" should _not_ be visible.
8. Clicking on **My Posts** should take you to a view in which you can see only the valid posts that were created by the current logged in user.
9. Clicking on any of the posts in either view should navigate you to the corresponding **Post Details** page.

Ticket #36, #43
